lib/presentation/same_type_events_builder/same_type_events_list_builder.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/presentation/same_type_events_builder/same_type_events_controller.dart';
+import 'package:fluffychat/widgets/twake_components/twake_loading/center_loading_indicator.dart';
+import 'package:flutter/material.dart';
+
+class SameTypeEventsBuilder extends StatelessWidget {
+  final SameTypeEventsBuilderController controller;
+
+  /// The builder must return a sliver.
+  final Widget Function(BuildContext, Either<Failure, Success>, Widget?)
+      builder;
+
+  const SameTypeEventsBuilder({
+    Key? key,
+    required this.controller,
+    required this.builder,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return CustomScrollView(
+      slivers: [
+        ValueListenableBuilder(
+          valueListenable: controller.refreshing,
+          builder: (context, refreshing, child) => SliverToBoxAdapter(
+            child:
+                refreshing ? const CenterLoadingIndicator() : const SizedBox(),
+          ),
+        ),
+        ValueListenableBuilder(
+          valueListenable: controller.eventsNotifier,
+          builder: builder,
+        ),
+        ValueListenableBuilder(
+          valueListenable: controller.loadingMore,
+          builder: (context, loadingMore, child) => SliverToBoxAdapter(
+            child:
+                loadingMore ? const CenterLoadingIndicator() : const SizedBox(),
```

Comment:
We use `SizedBox()`, `SizedBox.shrink()`, `Container()` in the application IMO we should be more consistent and choose one to make an empty widget.

Reply:
In this case i think we need use `SizedBox() or SizedBox.shrink()` because If you need a highly customizable widget with support for a wide range of layout properties, then Container is the way to go. But if you need a lightweight widget that can enforce a fixed size constraint or create a fixed-size gap between widgets, then SizedBox is the clear winner.